Marlin/Marlin_main.cpp

```diff
@@ -347,20 +347,31 @@ bool target_direction;
 #endif
 
 #ifdef DELTA
+
+  #define TOWER_1 X_AXIS
+  #define TOWER_2 Y_AXIS
+  #define TOWER_3 Z_AXIS
+
   float delta[3] = { 0 };
   #define SIN_60 0.8660254037844386
   #define COS_60 0.5
   float endstop_adj[3] = { 0 };
   // these are the default values, can be overriden with M665
   float delta_radius = DELTA_RADIUS;
-  float delta_tower1_x = -SIN_60 * delta_radius; // front left tower
-  float delta_tower1_y = -COS_60 * delta_radius;     
-  float delta_tower2_x =  SIN_60 * delta_radius; // front right tower
-  float delta_tower2_y = -COS_60 * delta_radius;     
+  float delta_tower1_x = -SIN_60 * (delta_radius + DELTA_RADIUS_TRIM_TOWER_1); // front left tower
+  float delta_tower1_y = -COS_60 * (delta_radius + DELTA_RADIUS_TRIM_TOWER_1);     
```

Comment:
Here you are adding a PER_TOWER CONSTANT to an adjustable radius.
I think that we should be more general and allow a variable adjustment to each tower.
Clearly, for the duration of a print, there results are invariant. Therefore they would be computed and stored in ram.

I would like to propose that each of the lengths be expressed in the following manner.
using the radius of one of towers as an example:

radius_tower_1 = TOWER_RADIUS + RADIUS_ADJUSTMENT_TOWER_1 + radius_trim_tower_1

The motivation is as follows:
Each length has an expected design value. If we know nothing else about the particular printer, that is a "good guess" for the actual value.
However, through some calibration procedure, we might learn a trim adjustment which we store in eprom (or on the host)
But some users would like to have the value in the firmware for that printer.
Here, they can define the ADJUSTMENTs. The compiler should be able to combine the DESIGN_VALUE with the adjustment into a single constant at compile time.
Then, show it be desired, the trim value could be made optional, thus saving space, etc. for those who wish to omit that feature.

Rather than having an ad-hoc way to handle each length, it would be good to handle ALL(radius, arm length, height, etc.) of the lengths in a uniform manner.